### PR TITLE
Fixed the two inhabitable type guarding overloads.

### PR DESCRIPTION
**Pull request by andersio**
These two `*Values(_:)* overloads were dropped under the wrong extension. They should be constrained with also `Error == NoError`.

#### Checklist
- ~~Updated CHANGELOG.md.~~

### CODE REVIEW

File: Sources/InhabitableTypeGuards.swift [hunk 1]
```diff
@@ -5,16 +5,16 @@ extension SignalProducer where Value == Never {
...
-	public func startWithValues(_ action: @escaping (Value) -> Void) -> Disposable { observingInhabitableTypeError() }
 }
 
 extension SignalProducer where Value == Never, Error == NoError {
 	@discardableResult
 	@available(*, deprecated, message:"Observer is never called - `Value` and `Error` are inhabitable (Instantiation at runtime would trap)")
 	public func startWithResult(_ action: @escaping (Result<Value, Error>) -> Void) -> Disposable { observingInhabitableTypeError() }
+
+	@discardableResult
+	@available(*, deprecated, message:"Observer is never called - `Value` is inhabitable (Instantiation at runtime would trap)")
```
Comment (sharplet) [inline Sources/InhabitableTypeGuards.swift]: Should this read "`Value` is <em>un</em>inhabitable"?

File: Sources/InhabitableTypeGuards.swift [hunk 2]
```diff
@@ -27,16 +27,16 @@ extension Signal where Value == Never {
...
 }
 
 extension Signal where Value == Never, Error == NoError {
 	@discardableResult
 	@available(*, deprecated, message:"Observer is never called - `Value` and `Error` are inhabitable (Instantiation at runtime would trap)")
 	public func observeResult(_ action: @escaping (Result<Value, Error>) -> Void) -> Disposable? { observingInhabitableTypeError() }
+	
+	@discardableResult
+	@available(*, deprecated, message:"Observer is never called - `Value` is inhabitable (Instantiation at runtime would trap)")
+	public func observeValues(_ action: @escaping (Value) -> Void) -> Disposable? { observingInhabitableTypeError() }
```
Comment (sharplet) [inline Sources/InhabitableTypeGuards.swift]: I think this should also be "uninhabitable".

**Comment by andersio** [comment]
The official word is _uninhabited_. 🙈

**Comment by andersio** [comment]
@sharplet @mdiep

**Pull request merged**